pkg/config/settings/runtime_setting_block_profile_rate.go

```diff
@@ -38,8 +38,10 @@ func (r RuntimeBlockProfileRate) Set(value interface{}) error {
 		return err
 	}
 
+	err = checkProfilingNeedsRestart(profiling.GetBlockProfileRate(), rate)
```

Comment:
The name `checkSomething..` suggests returning a boolean.  I see it's called twice, though, so it makes sense to write the error message once.  Maybe `verifyRunningProfileConfig`?